explorer/src/Pos/Explorer/Web/Transform.hs

```diff
@@ -3,36 +3,48 @@
 
 module Pos.Explorer.Web.Transform
        ( ExplorerProd
+       , runExplorerProd
+       , liftToExplorerProd
        , explorerServeWebReal
        , explorerPlugin
        , notifierPlugin
        ) where
 
 import           Universum
 
-import qualified Control.Monad.Catch     as Catch (Handler (..), catches)
-import           Control.Monad.Except    (MonadError (throwError))
-import qualified Control.Monad.Reader    as Mtl
-import           Mockable                (runProduction)
-import           Servant.Server          (Handler)
-import           Servant.Utils.Enter     ((:~>) (..), enter)
+import qualified Control.Monad.Catch       as Catch (Handler (..), catches)
+import           Control.Monad.Except      (MonadError (throwError))
+import qualified Control.Monad.Reader      as Mtl
+import           Mockable                  (runProduction)
+import           Servant.Server            (Handler)
+import           Servant.Utils.Enter       ((:~>) (..), enter)
 
-import           Pos.Communication       (OutSpecs, SendActions, WorkerSpec, worker)
-import           Pos.Core                (HasCoreConstants)
-import           Pos.Recovery            ()
-import           Pos.Ssc.GodTossing      (SscGodTossing)
-import           Pos.WorkMode            (RealMode, RealModeContext (..))
+import           Pos.Communication         (OutSpecs, SendActions, WorkerSpec, worker)
+import           Pos.Context               (NodeContext)
+import           Pos.Core                  (HasCoreConstants)
+import           Pos.Recovery              ()
+import           Pos.Ssc.GodTossing        (SscGodTossing)
+import           Pos.Util.Util             (lensOf)
+import           Pos.WorkMode              (RealMode, RealModeContext (..))
 
-import           Pos.Explorer            (ExplorerBListener, runExplorerBListener)
-import           Pos.Explorer.Socket.App (NotifierSettings, notifierApp)
-import           Pos.Explorer.Web.Server (explorerApp, explorerHandlers,
-                                          explorerServeImpl)
+import           Pos.Explorer              (ExplorerBListener, runExplorerBListener)
+import           Pos.Explorer.ExtraContext (ExtraContext, ExtraContextT, makeExtraCtx,
+                                            runExtraContextT)
+import           Pos.Explorer.Socket.App   (NotifierSettings, notifierApp)
+import           Pos.Explorer.Web.Server   (explorerApp, explorerHandlers,
+                                            explorerServeImpl)
 
 -----------------------------------------------------------------
 -- Transformation to `Handler`
 -----------------------------------------------------------------
 
-type ExplorerProd = ExplorerBListener (RealMode SscGodTossing)
+type ExplorerProd = ExtraContextT (ExplorerBListener (RealMode SscGodTossing))
```

Comment:
FPC's recommendation was not to use `ether`. The way to avoid `ether` here is to define `ExplorerContext` and use `ReaderT ExplorerContext Production` instead of `RealMode SscGodTossing`.
However, I don't mind using `ether`, so I don't insist on changing it. It's explorer after all :)

Reply:
:unamused: